server/graphql/common/expenses.ts

```diff
@@ -1152,16 +1129,50 @@ async function payExpenseWithPayPalAdaptive(remoteUser, expense, host, paymentMe
       toPaypalEmail,
       paymentMethod.token,
     );
-    await createTransactionFromPaidExpense(
-      host,
-      paymentMethod,
-      expense,
-      paymentResponse,
-      expense.UserId,
-      fees['paymentProcessorFeeInHostCurrency'],
-      fees['hostFeeInHostCurrency'],
-      fees['platformFeeInHostCurrency'],
+
+    const { createPaymentResponse, executePaymentResponse } = paymentResponse;
+
+    switch (executePaymentResponse.paymentExecStatus) {
+      case 'COMPLETED':
+        break;
+
+      case 'CREATED':
+        /*
+         * When we don't provide a preapprovalKey (paymentMethod.token) to payServices['paypal'](),
+         * it creates a payKey that we can use to redirect the user to PayPal.com to manually approve that payment
+         * TODO We should handle that case on the frontend
+         */
+        throw new errors.BadRequest(
+          `Please approve this payment manually on ${createPaymentResponse.paymentApprovalUrl}`,
+        );
+
+      case 'ERROR':
+        // Backward compatible error message parsing
+        // eslint-disable-next-line no-case-declarations
+        const errorMessage =
+          executePaymentResponse.payErrorList?.payError?.[0].error?.message ||
+          executePaymentResponse.payErrorList?.[0].error?.message;
+        throw new errors.ServerError(
+          `Error while paying the expense with PayPal: "${errorMessage}". Please contact support@opencollective.com or pay it manually through PayPal.`,
+        );
```

Comment:
I think we can wrap the case in a block rather than adding this es-lint exception. 🤔 

```suggestion
      case 'ERROR': {
        // Backward compatible error message parsing
        const errorMessage =
          executePaymentResponse.payErrorList?.payError?.[0].error?.message ||
          executePaymentResponse.payErrorList?.[0].error?.message;
        throw new errors.ServerError(
          `Error while paying the expense with PayPal: "${errorMessage}". Please contact support@opencollective.com or pay it manually through PayPal.`,
        );
      }
```

Reply:
As mentioned above, this code is just a cut/paste that I don't want to change now to reduce the risk of side effects and make this PR simpler to review. We can clean the code later.